src/lib.rs

```diff
@@ -60,6 +60,13 @@ pub struct Chunk<B: IntoBuf> {
 
 // ===== impl Body =====
 
+impl<B: IntoBuf> Body<B> {
+    pub fn is_empty(&self) -> bool {
+        // If the recv side is closed and the receive queue is empty, the body is empty.
+        self.inner.is_recv_eos()
```

Comment:
i think that's correct fsvo is_empty. I intend to expose "is there a body at all" and not "is there remaining body to be consumed".  i should probably rename this